qxf2_scheduler/routes.py

```diff
@@ -65,73 +68,89 @@ def index():
 
 
 @app.route("/interviewers")
-def listinterviewer():
-    "List the interviewer names,designation"
-    interviewer_work_time_slots = []
-    new_slot = Interviewers.query.join(Interviewertimeslots,Interviewers.interviewer_id==Interviewertimeslots.interviewer_id).values(Interviewers.interviewer_id,Interviewers.interviewer_email,Interviewertimeslots.interviewer_start_time,Interviewertimeslots.interviewer_end_time)
+def listinterviewers():
+    "List all the interviewer names"
+    all_interviewers = Interviewers.query.all()
+    my_interviewers_list = []
+    for each_interviewer in all_interviewers:
+        my_interviewers_list.append({'interviewer_id': each_interviewer.interviewer_id,
+                                     'interviewer_name': each_interviewer.interviewer_name})
+
+    return render_template("list-interviewers.html", result=my_interviewers_list)
+
+
+def parse_interviewer_detail(interviewer_details):
+    "Parse the interviewer detail with start and end time"    
+    time_dict = {}
+    time_dict['starttime'] = interviewer_details['interviewers_starttime']
+    time_dict['endtime'] = interviewer_details['interviewers_endtime']   
+    del interviewer_details['interviewers_starttime']
+    del interviewer_details['interviewers_endtime']
+    interviewer_details['time'] = time_dict
+
+    return interviewer_details
+
     
-    for interviewer_id,interviewer_email,interviewer_start_time,interviewer_end_time in new_slot:        
-        interviewer_details = {'interviewer_id':interviewer_id,'interviewer_email':interviewer_email,'interviewer_start_time':interviewer_start_time,
-        'interviewer_end_time':interviewer_end_time}        
-        if not interviewer_work_time_slots:            
-            interviewer_work_time_slots.append(interviewer_details)
+@app.route("/<interviewer_id>/interviewer/")
+def read_interviewer_details(interviewer_id):
+    "Displays all the interviewer details"
+    parsed_interviewer_details = []
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+    for each_detail in interviewer_details:
+        interviewer_detail = {
+            'interviewers_name': each_detail.interviewer_name,
+            'interviewers_email': each_detail.interviewer_email,
+            'interviewers_designation': each_detail.interviewer_designation,
+            'interviewers_starttime': each_detail.interviewer_start_time,
+            'interviewers_endtime': each_detail.interviewer_end_time}
+        
+        parsed_interviewer_detail = parse_interviewer_detail(interviewer_detail)
+        if not parsed_interviewer_details:
```

Comment:
For lists, it is clearer to check Len(list_name) == 0

Reply:
Done